exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/writer/BaseScalarWriter.java

```diff
@@ -211,7 +211,7 @@ protected boolean canExpand(int delta) {
 
   protected void overflowed() {
     if (listener == null) {
-      throw new IndexOutOfBoundsException("Overflow not supported");
+      throw new UnsupportedOperationException("Overflow not supported");
```

Comment:
Maybe change to `OversizedAllocationException`. See `FixedValueVectors.setInitialCapacity()`.

Reply:
I also considered `OversizedAllocationException` but IMO `UnsupportedOperationException` is a better fit.